include/geolib/ros/msg_conversions.h

```diff
@@ -42,6 +45,31 @@ inline void convert(const geo::Transform& t, geometry_msgs::Transform& msg) {
     convert(t.getBasis(), msg.rotation);
 }
 
+inline void convert(const TriangleI& t, shape_msgs::MeshTriangle& msg) {
+    msg.vertex_indices[0] = t.i1_;
+    msg.vertex_indices[1] = t.i2_;
+    msg.vertex_indices[2] = t.i3_;
+}
+
+void convert(const geo::Mesh& m, shape_msgs::Mesh& msg) {
+    std::vector<Vector3> points = m.getPoints();
+    std::vector<TriangleI> triangles = m.getTriangleIs();
+
+    for (std::vector<Vector3>::const_iterator it = points.begin(); it != points.end(); ++it)
+    {
+        geometry_msgs::Point point;
+        convert(*it, point);
+        msg.vertices.push_back(point);
+    }
+
+    for (std::vector<TriangleI>::const_iterator it = triangles.begin(); it != triangles.end(); ++it)
+    {
+        shape_msgs::MeshTriangle MeshTriangle;
+        convert(*it, MeshTriangle);
+        msg.triangles.push_back(MeshTriangle);
+    }
+}
```

Comment:
Out of scope, but the code looks silly. You'd really hope you can do this copying with a line like
```
return shape_msgs::Mesh(m.getPoints(), m.getTriangles());
```

Reply:
It is a ros message. These don't include fancy constructors

Reply:
Yep, it's a pity they don't generate such code.